code/modules/admin/battle_royale.dm

```diff
@@ -0,0 +1,432 @@
+	set name = "Battle Royale - Spawn Loot Drop (Minor)"
+	set category = "Event"
+	if(!check_rights(R_FUN))
+		to_chat(src, "<span class='warning'>You do not have permission to do that!</span>")
+		return
+	if(!GLOB.battle_royale)
+		to_chat(src, "<span class='warning'>No game is in progress.</span>")
+		return
+	GLOB.battle_royale.generate_good_drop()
+	log_admin("[key_name(usr)] generated a battle royale drop.")
+	message_admins("[key_name(usr)] generated a battle royale drop.")
+
+/client/proc/battle_royale_spawn_loot_good()
+	set name = "Battle Royale - Spawn Loot Drop (Major)"
+	set category = "Event"
+	if(!check_rights(R_FUN))
+		to_chat(src, "<span class='warning'>You do not have permission to do that!</span>")
+		return
+	if(!GLOB.battle_royale)
+		to_chat(src, "<span class='warning'>No game is in progress.</span>")
+		return
+	GLOB.battle_royale.generate_endgame_drop()
+	log_admin("[key_name(usr)] generated a good battle royale drop.")
+	message_admins("[key_name(usr)] generated a good battle royale drop.")
+
+/datum/battle_royale_controller
+	var/list/players
+	var/datum/proximity_monitor/advanced/battle_royale/field_wall
+	var/radius = 118
+	var/process_num = 0
+	var/list/death_wall
+	var/field_delay = 15
+	var/debug_mode = FALSE
+
+/datum/battle_royale_controller/Destroy(force, ...)
+	QDEL_LIST(death_wall)
+	for(var/client/C in GLOB.admins)
+		C.verbs -= BATTLE_ROYALE_AVERBS
+	. = ..()
+	GLOB.enter_allowed = TRUE
+	world.update_status()
+	GLOB.battle_royale = null
+
+//Trigger random events and shit, update the world border
+/datum/battle_royale_controller/process()
+	process_num++
+	//Once every 50 seconds
+	if(prob(2))
+		generate_basic_loot(2)
+	//Once every 100 seconds.
+	if(prob(1))
+		generate_good_drop()
+	var/living_victims = 0
+	var/mob/winner
+	for(var/mob/living/M as() in players)
+		if(QDELETED(M))
+			players -= M
+			continue
+		if(M.stat == DEAD)	//We aren't going to remove them from the list in case they somehow revive.
+			continue
+		var/turf/T = get_turf(M)
+		if(T.x > 128 + radius || T.x < 128 - radius || T.y > 128 + radius || T.y < 128 - radius)
+			to_chat(M, "<span class='warning'>You have left the zone!</span>")
+			M.gib()
+		if(!SSmapping.level_trait(T.z, ZTRAIT_STATION) && !SSmapping.level_trait(T.z, ZTRAIT_RESERVED))
+			to_chat(M, "<span class='warning'>You have left the z-level!</span>")
+			M.gib()
+		living_victims++
+		winner = M
+		CHECK_TICK
+	if(living_victims <= 1 && !debug_mode)
+		to_chat(world, "<span class='ratvar'><font size=18>VICTORY ROYALE!!</font></span>")
+		if(winner)
+			winner.client?.process_greentext()
+			to_chat(world, "<span class='ratvar'><font size=18>[key_name(winner)] is the winner!</font></span>")
+			new /obj/item/melee/supermatter_sword(get_turf(winner))
+		qdel(src)
+		return
+	//Once every 15 seconsd
+	// 1,920 seconds (about 32 minutes per game)
+	if(process_num % (field_delay) == 0)
+		for(var/obj/effect/death_wall/wall as() in death_wall)
```

Comment:
This is probably fairly laggy, wonder if you should CHECK_TICK it?

Reply:
Done